### PR TITLE
test: Add missing `set -ex` to ci/lint/06_script.sh

### PR DESCRIPTION
**Pull request by maflcko**
Requested in https://github.com/bitcoin/bitcoin/pull/28083#pullrequestreview-1535304219.

Also, one doc commit.

### CODE REVIEW

**Review by fanquake — APPROVED** [review]
ACK ffff4b5dc57c32bf759b705530c1368de4aa787e

File: ci/lint/06_script.sh [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 export LC_ALL=C
 
+set -ex
```
Comment (jamesob) [inline ci/lint/06_script.sh:9]: Only downside with this is that CI won't report all lint failures, rather will bail on the first one. Not sure if this is going to be annoying for people. Maybe we could have a `FAIL_FAST` envvar that gates this, and right now is only enabled in the container entrypoint script?

Comment (jamesob) [reply ci/lint/06_script.sh:9]: Nah, this is a step in the right direction I think.

**Review by jamesob — APPROVED** [review]
ACK ffff4b5dc57c32bf759b705530c1368de4aa787e ([`jamesob/ackr/28103.1.MarcoFalke.test_add_missing_set_ex`](https://github.com/jamesob/bitcoin/tree/ackr/28103.1.MarcoFalke.test_add_missing_set_ex))

Tested locally; lint errors surfaced through the lint container give non-zero error code as expected, which will allow local git hooks to e.g. stop pushes on error.

**Pull request merged**